### ISSUE TITLE
Fix S3253 FN: primary parameterless constructors

### ISSUE THREAD

**Issue by antonioaversa** [SonarSource/sonar-dotnet#8092]
Primary parameterless constructors are redundant as parameterless constructors and should be reported by [S3253](https://sonarsource.github.io/rspec/#/rspec/S3253).

```cs
class AClassWithBody() { }                // FN
class AClassWithoutBody();                // FN
struct AStructWithBody() { }              // FN
struct AStructWithoutBody();              // FN
record ARecordWithBody() { }              // FN
record ARecordWithoutBody();              // FN
record struct ARecordStructWithBody() { } // FN
record struct ARecordStructWithoutBody(); // FN
``` 

Alternative: extend [S3235](https://sonarsource.github.io/rspec/#/rspec/S3235) instead.
